core/src/main/scala/chisel3/Util.scala

```diff
@@ -0,0 +1,28 @@
+package chisel3
+
+private[chisel3] object Utils {
+
+  /** Return a "good" name from a string representing a Scala-generated Class name.
+    * @param the input class name
+    */
+  def goodName(className: String): String = {
+    /* A sequence of string filters applied to the name */
+    val filters: Seq[String => String] =
+      Seq(((a: String) => raw"\$$+anon".r.replaceAllIn(a, "_Anon")) // Merge the "$$anon" name with previous name
+      )
+
+    filters
+      .foldLeft(className) { case (str, filter) => filter(str) } // 1. Apply filters to baseName
+      .split("\\.|\\$") // 2. Split string at '.' or '$'
+      .filterNot(_.forall(_.isDigit)) // 3. Drop purely numeric names
+      .last // 4. Use the last name
+  }
+
+  /** Return a "good" class name for an object.  This is an improved version of `Class[_].getSimpleName`.
+    * @param obj any object
+    */
+  def goodName(obj: Any): String = {
+    goodName(obj.getClass().getName())
+  }
+
+}
```

Comment:
Just yesterday, @jared-barocsi added `chisel3.util.simpleClassName` (https://github.com/chipsalliance/chisel/blob/4186303c85e8c5afc60685485d2bc8090ffea5bc/core/src/main/scala/chisel3/util/Naming.scala#L6) defined in `core` which has this same implementation, you should use that instead 🙂